python/test/unit/test_event_generators.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# test_event_generators.py
+
+import unittest
+import numpy as np
+
+import arbor as arb
+
+# to be able to run .py file from child directory
+import sys, os
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../../')))
+
+try:
+    import options
+except ModuleNotFoundError:
+    from test import options
+
+"""
+all tests for event generators (regular, explicit, poisson)
+"""
+
+class RegularSchedule(unittest.TestCase):
+    def test_default_regular_schedule(self):
+        rs = arb.regular_schedule()
+        self.assertEqual(rs.tstart, np.finfo(np.float32).max)
+        self.assertEqual(rs.dt, 0)
+        self.assertEqual(rs.tstop, np.finfo(np.float32).max)
+        self.assertEqual(rs.tstart, rs.tstop)
+
+    def test_dt_contor_regular_schedule(self):
+        rs = arb.regular_schedule(0.025)
+        self.assertEqual(rs.tstart, 0)
+        self.assertEqual(float("{0:.3f}".format(rs.dt)), 0.025)
```

Comment:
Why do we cast like this to get the `dt` value?

Reply:
Done by using `assertAlmostEqual` (checking if `round(a-b, 7) == 0`).